flatdata-rs/lib/src/arrayview.rs

```diff
@@ -1,242 +1,47 @@
-use crate::{
-    structs::{RefFactory, Struct},
-    vector::Vector,
-};
+use crate::error::ResourceStorageError;
+use crate::structs::Struct;
 
-use std::{
-    fmt, iter, marker,
-    ops::{Bound, RangeBounds},
-};
-
-/// A read-only view on a contiguous sequence of flatdata structs of the same
-/// type `T`.
-///
-/// The sequence is written using [`Vector`] or [`ExternalVector`]. For detailed
-/// examples see either of the two.
-///
-/// An archive provides a getter for each vector resource, which returns an
-/// array view.
-///
-/// [`Vector`]: struct.Vector.html
-/// [`ExternalVector`]: struct.ExternalVector.html
-#[derive(Clone)]
-pub struct ArrayView<'a, T>
-where
-    T: RefFactory,
-{
-    data: &'a [u8],
-    _phantom: marker::PhantomData<T>,
-}
-
-impl<'a, T> ArrayView<'a, T>
+/// Enhanced slices of flatdata Structs so that they can be created from bytes / converted to bytes
+pub trait SliceExt<'a>
```

Comment:
Just an observation: Looks like this is `TryFrom<&[u8]> + AsRef<[u8]>`. However, I also prefer explicit names.

Reply:
Problem is that you cannot implement AsRef/TryFrom (I tried): Both the trait, as well as the type (slice) are foreign.

Reply:
Thanks. Good to know.